platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AttributionDialogManager.java

```diff
@@ -0,0 +1,169 @@
+package com.mapbox.mapboxsdk.maps;
+
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AlertDialog;
+import android.text.Html;
+import android.text.SpannableStringBuilder;
+import android.text.TextUtils;
+import android.text.style.URLSpan;
+import android.view.View;
+import android.widget.ArrayAdapter;
+
+import com.mapbox.mapboxsdk.R;
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.style.sources.Source;
+import com.mapbox.services.android.telemetry.MapboxTelemetry;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Locale;
+
+/**
+ * Responsible for managing attribution interactions on the map.
+ * <p>
+ * When the user clicks the attribution icon, {@link AttributionDialogManager#onClick(View)} will be invoked.
+ * An attribution dialog will be shown to the user with contents based on the attributions found in the map style.
+ * Additionally an telemetry option item is shown to configure telemetry settings.
+ * </p>
+ */
+class AttributionDialogManager implements View.OnClickListener, DialogInterface.OnClickListener {
+
+  private static final String MAP_FEEDBACK_URL = "https://www.mapbox.com/map-feedback";
```

Comment:
We shouldn’t hard-code this URL because sources other than Mapbox Streets are free to provide their own map feedback URLs. Look for `<a>` tags with `class` set to `mapbox-improve-map`.